lib/protobuf/field/enum_field.rb

```diff
@@ -32,30 +32,18 @@ def enum?
         true
       end
 
+      def coerce!(value)
+        enum_value = type_class.fetch(value)
+        fail TypeError, "Invalid Enum value: #{value.inspect} for #{name}" unless enum_value
```

Comment:
Is `type_class` a hash and it's calling `fetch` on the hash, or is it a custom class? If it's the standard implementation of `Hash#fetch`, this wouldn't ever do something unless somehow an enum value can be nil.